docs/source/proposals/external-memory-management.rst

```diff
@@ -0,0 +1,918 @@
+           # Allocates device memory using RMM functions. The finalizer for the
+           # allocated memory calls back to RMM to free the memory.
+           addr = librmm.rmm_alloc(bytesize, 0)
+           ctx = cuda.current_context()
+           ptr = ctypes.c_uint64(int(addr))
+           finalizer = _make_finalizer(addr, stream)
+           return MemoryPointer(ctx, ptr, bytesize, finalizer=finalizer)
+
+      def get_ipc_handle(self, memory):
+           """ 
+           Get an IPC handle for the memory with offset modified by the RMM memory
+           pool.
+           """
+           # This implementation provides a functional implementation and illustrates
+           # what get_ipc_handle needs to do, but it is not a very "clean"
+           # implementation, and it relies on borrowing bits of Numba internals to
+           # initialise ipchandle. 
+           #
+           # A more polished implementation might make use of additional functions in
+           # the RMM C++ layer for initialising IPC handles, and not use any Numba
+           # internals.
+           ipchandle = (ctypes.c_byte * 64)()  # IPC handle is 64 bytes
+           cuda.cudadrv.memory.driver_funcs.cuIpcGetMemHandle(
+               ctypes.byref(ipchandle),
+               memory.owner.handle,
+           )
+           source_info = cuda.current_context().device.get_device_identity()
+           ptr = memory.device_ctypes_pointer.value
+           offset = librmm.rmm_getallocationoffset(ptr, 0)
+           return IpcHandle(memory, ipchandle, memory.size, source_info,
+                            offset=offset)
+
+       def get_memory_info(self):
+           # Returns a tuple of (free, total) using RMM functionality.
+           return get_memory_info()
+
+       def initialize(self):
+           # Nothing required to initialize RMM here, but this method is added
+           # to illustrate that the super() method should also be called.
+           super().initialize()
+
+       @contextmanager
+       def defer_cleanup(self):
+           # Does nothing to defer cleanup - a full implementation may choose to
+           # implement a different policy.
+           with super().defer_cleanup():
+               yield
+
+       @property
+       def interface_version(self):
+           # As required by the specification
+           return 1
+
+   # The existing _make_finalizer function is used by RMMNumbaManager:
+   def _make_finalizer(handle, stream):
+       """
+       Factory to make the finalizer function.
+       We need to bind *handle* and *stream* into the actual finalizer, which
+       takes no args.
+       """
+
+       def finalizer():
+           """
+           Invoked when the MemoryPointer is freed
+           """
+           librmm.rmm_free(handle, stream)
+
+       return finalizer
+
+   # Utility function register `RMMNumbaManager` as an EMM:
+   def use_rmm_for_numba():
+       cuda.set_memory_manager(RMMNumbaManager)
```

Comment:
The one already imported by rmm.py. For clarity though, I've just changed it so that `set_memory_manager` is explicitly imported so it can be seen as a new addition.